compiler/luci/pass/src/QuantizeDequantizeWeightsPass.cpp

```diff
@@ -70,6 +70,210 @@ void compute_asym_scale_zp(float min, float max, float &scaling_factor, int64_t
   zp = nudged_zero_point;
 }
 
+bool get_channel_dim_index(CircleConst *node, int *len_per_dim, int &channel_dim_index)
+{
+  auto succs = loco::succs(node);
+  if (succs.size() != 1) // assume weights is used by only one node
+    return false;
+
+  for (auto out : succs)
+  {
+    auto conv = dynamic_cast<CircleConv2D *>(out);
+    auto dw_conv = dynamic_cast<CircleDepthwiseConv2D *>(out);
+    auto tw_conv = dynamic_cast<CircleTransposeConv *>(out);
+    auto fc = dynamic_cast<CircleFullyConnected *>(out);
+
+    if (conv != nullptr && conv->filter() == node)
+    {
+      len_per_dim[0] = node->dim(0).value();
```

Comment:
What we discussed off-line about checking `node->rank() == 4`,
there is `bool is_weights(CircleNode *node)` method that does conditional check.
I think it would be better to check `node->rank() == 4` for `CircleConv2D ` in `is_weights()`...
@jinevening , what do you think?

Reply:
> @jinevening , what do you think?

I didn't hear the off-line discussion.. I think the rank of CircleConv**2D** must be 4 by its definition. It should be checked when the node is created (and the rank should not be changed). AFAIK, there is no such guarantee, so it would be ok to add assertion to check the rank here.

Reply:
> Adding assert(node->rank() == 4) is needed for safety.

I agree that it will make the code safer than before. Can you specify the line number where the assertion is inserted or make suggestions? It would be very helpful for @meejeong .

Reply:
I think this can make clear?
1) add `if (node->rank() != 4) return false` in `is_weights()` method
2) add `assert(node->rank() == 4)` here, line 88.